From 1a5b148a9be4a4947a691351dc694c6e265060b1 Mon Sep 17 00:00:00 2001
From: Joey Hess <joey@kitenet.net>
Date: Thu, 16 Oct 2014 12:29:25 -0400
Subject: [PATCH] void using of chmod on android when changing config

This breaks on Android's /sdcard, which has a variety of FUSE
implentations, all total shite.
---
 config.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/config.c b/config.c
index 15a2983..cb0db4d 100644
--- a/config.c
+++ b/config.c
@@ -2041,12 +2041,14 @@ int git_config_set_multivar_in_file(const char *config_filename,
 			MAP_PRIVATE, in_fd, 0);
 		close(in_fd);
 
+		/* not on android
 		if (chmod(lock->filename.buf, st.st_mode & 07777) < 0) {
 			error("chmod on %s failed: %s",
 				lock->filename.buf, strerror(errno));
 			ret = CONFIG_NO_WRITE;
 			goto out_free;
 		}
+		*/
 
 		if (store.seen == 0)
 			store.seen = 1;
@@ -2224,11 +2226,13 @@ int git_config_rename_section_in_file(const char *config_filename,
 
 	fstat(fileno(config_file), &st);
 
+	/* not on android
 	if (chmod(lock->filename.buf, st.st_mode & 07777) < 0) {
 		ret = error("chmod on %s failed: %s",
 				lock->filename.buf, strerror(errno));
 		goto out;
 	}
+	*/
 
 	while (fgets(buf, sizeof(buf), config_file)) {
 		int i;
-- 
2.1.1

